marklogic-data-hub/src/main/java/com/marklogic/hub/deploy/commands/LoadUserModulesCommand.java

```diff
@@ -270,6 +270,12 @@ public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) {
 
     public void setHubConfig(HubConfig hubConfig) {
         this.hubConfig = hubConfig;
+        if (this.entityManager == null) {
+            this.entityManager = new EntityManagerImpl();
+        }
+        if (this.entityManager instanceof EntityManagerImpl) {
```

Comment:
EntityManagerImpl is the only implementation of EntityManager. Maybe we should add the setHubConfig as an new interface method instead of checking the instance type and do the cast...

Reply:
Setters for dependencies shouldn't be part of an interface, as that exposes the implementation. EntityManagerImpl(HubConfig) should be used instead when setHubConfig is called (though I don't think there should be a setHubConfig on LoadUserModulesCommand either - it should be LoadUserModulesCommand(HubConfig) instead).